src/haddock/modules/scoring/sasascore/__init__.py

```diff
@@ -78,12 +58,14 @@ def _run(self) -> None:
                 )
         except Exception as e:
             self.finish_with_error(e)
-
-        nmodels = len(models_to_score)
-        # index_list for the jobs with linear scaling
-        ncores = parse_ncores(n=self.params['ncores'], njobs=nmodels)
-        idx_list = get_index_list(nmodels, ncores)
-
+        if self.params["mode"] == "mpi":
+            input_ncores_slurm = int(os.getenv('SLURM_CPUS_ON_NODE', 1))
+            input_ncores_pbs = int(os.getenv('PBS_CPUS_ON_NODE', 1))
+            input_ncores = max(input_ncores_slurm, input_ncores_pbs)
+        else:
+            input_ncores = self.params["ncores"]
```

Comment:
Don't we want to have this machinery set in `libs/libparallel.py` ?

So it can be used by other modules.

```python3
def get_input_ncores(params: ParamDict) -> int:
    if self.params["mode"] == "mpi":
        input_ncores_slurm = int(os.getenv('SLURM_CPUS_ON_NODE', 1))
        input_ncores_pbs = int(os.getenv('PBS_CPUS_ON_NODE', 1))
        input_ncores = max(input_ncores_slurm, input_ncores_pbs)
    else:
        input_ncores = self.params["ncores"]
    return input_ncores
```

Reply:
of course, but in another PR (if we agree on this implementation)

Reply:
It should actually be in the `ScoringModule` class:

```python
class ScoringModule(BaseHaddockModule):
    """Parent class for Scoring modules."""

    def __init__(self, order: int, path: Path, initial_params: FilePath) -> None:
        super().__init__(order, path, initial_params)

        if self.params["mode"] == "mpi":
            input_ncores_slurm = int(os.getenv("SLURM_CPUS_ON_NODE", 1))
            input_ncores_pbs = int(os.getenv("PBS_CPUS_ON_NODE", 1))
            self.params["ncores"] = max(input_ncores_slurm, input_ncores_pbs)
```

Reply:
Actually since this affects all modules it should be lower, most likely inside `parse_cores`